src/components/Sidebar/index.js

```diff
@@ -1,56 +1,149 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Swal from 'sweetalert2';
+
+import logo from 'assets/images/logo.svg';
+import {
+  allSidebarProjects,
+  allUpdateProject,
+  sampleProjects,
+} from 'store/actions/project';
 
 import './style.scss';
 
 function Sidebar() {
+  const dispatch = useDispatch();
+
+  const allState = useSelector((state) => state);
+
+  const [myProjects, setMyProjects] = useState([]);
+  const [sampleProject, setSampleProjects] = useState([]);
+  const [updateProject, setUpdateProject] = useState([]);
+
+  useEffect(() => {
+    if (allState.sidebar.allProject.length === 0) {
+      dispatch(allSidebarProjects());
+      dispatch(sampleProjects());
+      dispatch(allUpdateProject());
+    }
+  }, []);
+
+  useEffect(() => {
+    if (allState.sidebar.allProject.length > 0) {
+      setMyProjects(allState.sidebar.allProject);
+    }
+  }, [allState.sidebar.allProject]);
+
+  useEffect(() => {
+    if (allState.sidebar.sampleProject.length > 0) {
+      setSampleProjects(allState.sidebar.sampleProject);
+    }
+  }, [allState.sidebar.sampleProject]);
+
+  useEffect(() => {
+    if (allState.sidebar.updateProject.length > 0) {
+      setUpdateProject(allState.sidebar.updateProject);
+    }
+  }, [allState.sidebar.updateProject]);
```

Comment:
It seems like we can use allProject, sampleProject and updateProject from store directly without creating state
There is no modification after assignment

Reply:
this approach is more safer while dealing with API and store.  our plan is to add more data in sidebar.

Reply:
Add more data in sidebar component? or globally?

Reply:
We will add teams in sidebar. and they will toggle on the base of active or online members in the future